backend/users/views.py

```diff
@@ -243,3 +244,151 @@ def change_password_view(request):
         return render(request, "users/change_password.html", {"error": error})
 
     return render(request, "users/change_password.html")
+
+
+##############
+##### Friends Methods
+##############
+
+
+@jwt_login_required
+def add_friend_view(request):
+    error_message = None
+
+    if request.method == "POST":
+        username = request.POST.get("username")
+        friend = CustomUser.objects.filter(username=username).first()  # Use filter to avoid 404
+
+        if not friend:
+            error_message = "User not found."
+        else:
+            request.user.add_friend(friend)
+
+        if error_message:
+            return render(request, "users/friend_list.html", {"error_message": error_message})
+
+        return redirect("friend_list")  # Redirect to the friends list after adding
+
+    return render(request, "users/add_friend.html")
+    
+@jwt_login_required
+def add_friend(request, username):
+    friend = CustomUser.objects.filter(username=username).first()
+    error_message = None
+
+    if not friend:
+        error_message = "User not found."
+    elif friend == request.user:
+        error_message = "You cannot add yourself as a friend."
+    else:
+        request.user.add_friend(friend)
+
+    if error_message:
+        return render(request, "users/friend_list.html", {"error_message": error_message})
+
+    return redirect("friend_list")
+
+
+@jwt_login_required
+def remove_friend(request, username):
```

Comment:
Is there a button to remove friend on the frontend?

Reply:
done